src/scripts/main.js

```diff
@@ -1,3 +1,18 @@
 'use strict';
 
 // write code here
+
+const thead = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+
+const th = Array.from(document.querySelectorAll('thead > tr > th'));
+
+thead.addEventListener('click', (e) => {
+  const index = th.indexOf(e.target);
+
+  const rowSort = [...tbody.rows].sort((a, b) => {
+    return a.children[index].innerText - b.children[index].innerText;
```

Comment:
The current sorting logic uses subtraction, which is only suitable for numeric values. If the table contains strings or other data types, this will not work as expected. Consider using `localeCompare` for string comparison or implementing a more robust sorting function that can handle different data types.